samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -280,4 +376,68 @@ private void setApplicationFinalStatus() {
       }
     }
   }
+
+
+  /**
+   * Defines a specific implementation of {@link DistributedDataStateListener} for local {@link DistributedDataAccess}
+   */
+  private final class LocalDistributedDataStateListener implements DistributedDataStateListener {
+
+    /**
+     * upon reconnect check if global runid differs from local runid
+     */
+    @Override
+    public void handleReconnect() {
+      if(coordinationUtils == null || runIdLock == null || runIdAccess == null ) {
+        LOG.warn("Stopping processor {} and shutting down due to failure reading global runid after reconnect", uid);
```

Comment:
where is it reading the global runid?

Reply:
actually meant to say it could not read global runid. HOwever, this code is now moved out of LAR to make it cleaner